I suspect i hit this today but the message is slightly different so here goes:
The cli (v 0.4.40) appears to start fine, but after starting a model (in this case gemini 2.5 pro), it fails to run simple bash commands like 'mkdir'. I tried clearing cache folder and restarting, but same error. My humble request: **please** do some kind of self-test to detect the issue in startup before the model does - I burned a few $$ today while the model built up context reading files, but then it couldn't do anything with it!

Bash Build the demo app                                                                                                                                                                     ┃  ResolveMessage: Cannot find module '/$bunfs/root/../../../tree-sitter-d8jpqavy.node' from '/$bunfs/root/opencode'                                                                            

NixOS on aarch64.

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -17,7 +17,7 @@ jobs:
 
       - uses: oven-sh/setup-bun@v1
         with:
-          bun-version: 1.2.17
+          bun-version: 1.2.19
 
       - run: bun install
 

--- a/.github/workflows/publish-vscode.yml
+++ b/.github/workflows/publish-vscode.yml
@@ -21,7 +21,7 @@ jobs:
 
       - uses: oven-sh/setup-bun@v2
         with:
-          bun-version: 1.2.17
+          bun-version: 1.2.19
 
       - run: git fetch --force --tags
       - run: bun install -g @vscode/vsce

--- a/package.json
+++ b/package.json
@@ -3,7 +3,7 @@
   "name": "opencode",
   "private": true,
   "type": "module",
-  "packageManager": "bun@1.2.14",
+  "packageManager": "bun@1.2.19",
   "scripts": {
     "dev": "bun run --conditions=development packages/opencode/src/index.ts",
     "typecheck": "bun run --filter='*' typecheck",
@@ -46,7 +46,10 @@
   "trustedDependencies": [
     "esbuild",
     "protobufjs",
-    "sharp"
+    "sharp",
+    "tree-sitter",
+    "tree-sitter-bash",
+    "web-tree-sitter"
   ],
   "patchedDependencies": {}
 }

--- a/packages/opencode/package.json
+++ b/packages/opencode/package.json
@@ -52,6 +52,7 @@
     "remeda": "catalog:",
     "tree-sitter": "0.22.4",
     "tree-sitter-bash": "0.23.3",
+    "web-tree-sitter": "0.22.6",
     "turndown": "7.2.0",
     "vscode-jsonrpc": "8.2.1",
     "xdg-basedir": "5.1.0",

--- a/packages/opencode/src/tool/bash.ts
+++ b/packages/opencode/src/tool/bash.ts
@@ -19,11 +19,28 @@ const MAX_TIMEOUT = 10 * 60 * 1000
 const log = Log.create({ service: "bash-tool" })
 
 const parser = lazy(async () => {
-  const { default: Parser } = await import("tree-sitter")
-  const Bash = await import("tree-sitter-bash")
-  const p = new Parser()
-  p.setLanguage(Bash.language as any)
-  return p
+  try {
+    const { default: Parser } = await import("tree-sitter")
+    const Bash = await import("tree-sitter-bash")
+    const p = new Parser()
+    p.setLanguage(Bash.language as any)
+    return p
+  } catch (e) {
+    const { default: Parser } = await import("web-tree-sitter")
+    const { default: treeWasm } = await import("web-tree-sitter/tree-sitter.wasm" as string, { with: { type: "wasm" } })
+    await Parser.init({
+      locateFile() {
+        return treeWasm
+      },
+    })
+    const { default: bashWasm } = await import("tree-sitter-bash/tree-sitter-bash.wasm" as string, {
+      with: { type: "wasm" },
+    })
+    const bashLanguage = await Parser.Language.load(bashWasm)
+    const p = new Parser()
+    p.setLanguage(bashLanguage)
+    return p
+  }
 })
 
 export const BashTool = Tool.define("bash", {

--- a/packages/opencode/src/tool/test.ts
+++ b/packages/opencode/src/tool/test.ts
@@ -1,53 +1,70 @@
-import Parser from "tree-sitter";
-import Bash from "tree-sitter-bash";
+const parser = async () => {
+  try {
+    const { default: Parser } = await import("tree-sitter")
+    const Bash = await import("tree-sitter-bash")
+    const p = new Parser()
+    p.setLanguage(Bash.language as any)
+    return p
+  } catch (e) {
+    const { default: Parser } = await import("web-tree-sitter")
+    const { default: treeWasm } = await import("web-tree-sitter/tree-sitter.wasm" as string, { with: { type: "wasm" } })
+    await Parser.init({
+      locateFile() {
+        return treeWasm
+      },
+    })
+    const { default: bashWasm } = await import("tree-sitter-bash/tree-sitter-bash.wasm" as string, {
+      with: { type: "wasm" },
+    })
+    const bashLanguage = await Parser.Language.load(bashWasm)
+    const p = new Parser()
+    p.setLanguage(bashLanguage)
+    return p
+  }
+}
 
-const parser = new Parser();
-parser.setLanguage(Bash.language as any);
+const sourceCode = `cd --foo foo/bar && echo "hello" && cd ../baz`
 
-const sourceCode = `cd --foo foo/bar && echo "hello" && cd ../baz`;
-
-const tree = parser.parse(sourceCode);
+const tree = await parser().then((p) => p.parse(sourceCode))
 
 // Function to extract commands and arguments
-function extractCommands(
-  node: any,
-): Array<{ command: string; args: string[] }> {
-  const commands: Array<{ command: string; args: string[] }> = [];
+function extractCommands(node: any): Array<{ command: string; args: string[] }> {
+  const commands: Array<{ command: string; args: string[] }> = []
 
   function traverse(node: any) {
     if (node.type === "command") {
-      const commandNode = node.child(0);
+      const commandNode = node.child(0)
       if (commandNode) {
-        const command = commandNode.text;
-        const args: string[] = [];
+        const command = commandNode.text
+        const args: string[] = []
 
         // Extract arguments
         for (let i = 1; i < node.childCount; i++) {
-          const child = node.child(i);
+          const child = node.child(i)
           if (child && child.type === "word") {
-            args.push(child.text);
+            args.push(child.text)
           }
         }
 
-        commands.push({ command, args });
+        commands.push({ command, args })
       }
     }
 
     // Traverse children
     for (let i = 0; i < node.childCount; i++) {
-      traverse(node.child(i));
+      traverse(node.child(i))
     }
   }
 
-  traverse(node);
-  return commands;
+  traverse(node)
+  return commands
 }
 
 // Extract and display commands
-console.log("Source code: " + sourceCode);
-const commands = extractCommands(tree.rootNode);
-console.log("Extracted commands:");
+console.log("Source code: " + sourceCode)
+const commands = extractCommands(tree.rootNode)
+console.log("Extracted commands:")
 commands.forEach((cmd, index) => {
-  console.log(`${index + 1}. Command: ${cmd.command}`);
-  console.log(`   Args: [${cmd.args.join(", ")}]`);
-});
+  console.log(`${index + 1}. Command: ${cmd.command}`)
+  console.log(`   Args: [${cmd.args.join(", ")}]`)
+})

